modules/gapi/test/common/gapi_video_tests_common.hpp

```diff
@@ -0,0 +1,179 @@
+                                                        FloatVector&>& outOCVTuple,
+                                             std::tuple<Point2fVector&,
+                                                        UcharVector&,
+                                                        FloatVector&>& outGAPITuple)
+{
+    testInst.initMatsFromImages(testParams.channels,
+                                testParams.fileNamePattern,
+                                testParams.format);
+
+    auto inTuple = std::make_tuple(std::ref(testInst.in_mat1), std::ref(testInst.in_mat2),
+                                   std::ref(inPts));
+    return runOCVnGAPIOptFlowLK<cv::GMat>(static_cast<size_t>(testInst.in_mat1.cols),
+                                          static_cast<size_t>(testInst.in_mat1.rows),
+                                          testParams,
+                                          inTuple, outOCVTuple, outGAPITuple);
+}
+
+inline cv::GComputation runOCVnGAPIOptFlowLKForPyr(TestFunctional& testInst,
+                                                   const OptFlowLKTestParams& testParams,
+                                                   bool withDeriv,
+                                                   std::tuple<MatVector&,
+                                                              MatVector&,
+                                                              Point2fVector&>& inTuple,
+                                                   std::tuple<Point2fVector&,
+                                                              UcharVector&,
+                                                              FloatVector&>& outOCVTuple,
+                                                   std::tuple<Point2fVector&,
+                                                              UcharVector&,
+                                                              FloatVector&>& outGAPITuple)
+{
+    testInst.initMatsFromImages(testParams.channels,
+                                testParams.fileNamePattern,
+                                testParams.format);
+
+    cv::Size winSize(testParams.winSize, testParams.winSize);
+
+    int maxLevel;
+    maxLevel = cv::buildOpticalFlowPyramid(testInst.in_mat1, std::get<0>(inTuple),
+                                           winSize, testParams.maxLevel, withDeriv);
+    maxLevel = cv::buildOpticalFlowPyramid(testInst.in_mat2, std::get<1>(inTuple),
+                                           winSize, testParams.maxLevel, withDeriv);
+
+    OptFlowLKTestParams updatedTestParams(testParams);
+    updatedTestParams.maxLevel = maxLevel;
+
+    return runOCVnGAPIOptFlowLK<cgv::GGMatArray>(static_cast<size_t>(testInst.in_mat1.cols),
+                                                 static_cast<size_t>(testInst.in_mat1.rows),
+                                                 updatedTestParams,
+                                                 inTuple, outOCVTuple, outGAPITuple);
+}
+} // namespace
```

Comment:
I believe this code may be more compact. I assume we save already by reusing this function in every test, but still this function itself looks scary.

Reply:
A try has been done.